### ISSUE TITLE
Issue when linking Oracle Call Interface

### ISSUE THREAD

**Issue by tqshay** [edenhill/librdkafka#696]
Description
===========

I am having issues getting this library to work properly whenever I also link the Oracle Call Interface.  When OCI is linked, I continually get a message saying "Can't open/find Kafka configuration file".

This is what I have been using to compile:
cc -v -g -xCC -m64 -xarch=native -xcode=pic32 -mt -DSOLARIS -DM_UNIX -DLIMITED -D_LARGEFILE64_SOURCE=1 -DBIT64 -xstrconst -I/transform/development/base/kafka/include/librdkafka rob_kafka.c -o ../bin/rob_kafka -lsocket -lnsl -lpthread -lz -lcrypto -lssl -lsasl -lrt -L/transform/development/base/libso -lrdkafka -L/xf/xform/app/oracle/product/11.2.0.4/lib/ -lclntsh

When "-L/xf/xform/app/oracle/product/11.2.0.4/lib/ -lclntsh" is left off, the code compiles without issue.  Unfortunately, I need the OCI for my database connection.  Is there a known issue between librdkafka and OCI?  Is there a workaround?

How to reproduce
================

Checklist
=========
Please provide the following information:

 - [ ] librdkafka version (release number or git tag):
 - [ ] Apache Kafka version: 
 - [ ] librdkafka client configuration:
 - [ ] Operating system:
 - [ ] Using the legacy Consumer
 - [ ] Using the high-level KafkaConsumer
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by tqshay** [comment]
In addition, I have attempted compiling with gcc.  I have the same issue.

This does not compile:
/opt/csw/bin/gcc -m64 -mcpu=v9 -D_XPG6 -g -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT -D__EXTENSIONS__ -I/transform/development/base/kafka/include/librdkafka rob_kafka.c -o ../bin/rob_kafka -lsocket -lnsl -lpthread -lz -lcrypto -lssl -lsasl -lrt -L/transform/development/base/libso -lrdkafka -L/xf/xform/app/oracle/product/11.2.0.4/lib/ -lclntsh

This does compile:
/opt/csw/bin/gcc -m64 -mcpu=v9 -D_XPG6 -g -O2 -fPIC -Wall -Wsign-compare -Wfloat-equal -Wpointer-arith -D_POSIX_PTHREAD_SEMANTICS -D_REENTRANT -D__EXTENSIONS__ -I/transform/development/base/kafka/include/librdkafka rob_kafka.c -o ../bin/rob_kafka -lsocket -lnsl -lpthread -lz -lcrypto -lssl -lsasl -lrt -L/transform/development/base/libso -lrdkafka

**Comment by edenhill** [comment]
Hello,

the log message `"Can't open/find Kafka configuration file"` is not coming from librdkafka, some other component must be logging that.

The only other thing that sticks out is the linking order, try linking -lrdkafka before all the other libraries.

**Comment by tqshay** [comment]
So... We finally figured out what was going on.  There is an incompatibility between the Oracle 11.2.0.2 / 11.2.0.4 libraries and librdkafka.  Whenever we linked either of these versions, we saw the behavior I mentioned above.  When we linked an older version of Oracle (10.2.0), our issue magically went away.
We have put in a ticket with Oracle for them to investigate why their library is causing this sort of issue.
